apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx

```diff
@@ -1,25 +1,129 @@
     functionDetails: SuiMoveNormalizedFunction;
     defaultOpen?: boolean;
 };
+
 export function ModuleFunction({
     defaultOpen,
+    packageId,
+    moduleName,
     functionName,
     functionDetails,
 }: ModuleFunctionProps) {
+    const { connected, signAndExecuteTransaction } = useWallet();
+    const allParamsDetails = useFunctionParamsDetails(
+        functionDetails.parameters
+    );
+    const filteredParamsDetails = useMemo(
+        () => allParamsDetails.filter((aParam) => !aParam.isTxContext),
+        [allParamsDetails]
+    );
+    const defaultValues = useMemo(
+        () =>
+            Array.from({ length: filteredParamsDetails.length }, () => ({
+                value: '',
+            })),
+        [filteredParamsDetails.length]
+    );
+    const { register, handleSubmit, formState, control } = useZodForm(
+        argsSchema,
+        {
+            defaultValues: {
+                params: defaultValues,
+            },
+        }
+    );
+    const { fields } = useFieldArray({ control, name: 'params' });
+    const isExecuteDisabled =
+        formState.isValidating ||
+        !formState.isValid ||
+        formState.isSubmitting ||
+        !connected;
     return (
         <DisclosureBox defaultOpen={defaultOpen} title={functionName}>
-            <pre>{JSON.stringify(functionDetails, null, 2)}</pre>
+            <form
+                onSubmit={handleSubmit(async ({ params }) => {
+                    try {
+                        await toast.promise(
+                            signAndExecuteTransaction({
+                                kind: 'moveCall',
+                                data: {
+                                    packageObjectId: packageId,
+                                    module: moduleName,
+                                    function: functionName,
+                                    arguments: params.map(({ value }) => value),
+                                    typeArguments: [], // TODO: currently move calls that expect type argument will fail
+                                    gasBudget: 2000,
+                                },
+                            }).then((tx) => {
+                                if (tx.effects.status.status === 'failure') {
+                                    throw new Error(
+                                        tx.effects.status.error ||
+                                            'Transaction failed'
+                                    );
+                                }
+                            }),
+                            {
+                                loading: 'Executing...',
+                                error: (e) => 'Transaction failed',
+                                success: 'Done',
+                            }
+                        );
+                    } catch (e) {}
+                })}
+                autoComplete="off"
+                className="flex flex-col flex-nowrap items-stretch gap-3.75"
```

Comment:
Can we just use 4 instead of 3.75? I really want to try to stick to a consistent spacing scale.